WordPress/Sniffs/Theme/NoDeregisterCoreScriptSniff.php

```diff
@@ -0,0 +1,82 @@
+<?php
+/**
+ * Sniff to check theme for deregister and register of core scripts (jquery)
+ *
+ * @category Theme
+ * @package  PHP_CodeSniffer
+ * @author   Simon Prosser <pross@pross.org.uk>
+ */
+class WordPress_Sniffs_Theme_NoDeregisterCoreScriptSniff implements PHP_CodeSniffer_Sniff {
+
+	/**
+	 * Core scripts to sniff for.
+	 *
+	 * @var array
+	 */
+	private $core_scripts = array(
+		'jquery',
+	);
+
+	/**
+	 * Returns an array of tokens this test wants to listen for.
+	 *
+	 * @return array
+	 */
+	public function register() {
+		return array(
+			T_STRING,
+			T_CONSTANT_ENCAPSED_STRING,
+			T_DOUBLE_QUOTED_STRING,
+			T_OPEN_PARENTHESIS,
+			T_CLOSE_PARENTHESIS,
+		);
+	}//end register()
+
+	/**
+	 * Process a given string and remove quotes.
+	 *
+	 * @param string $string The text to be processed.
+	 *
+	 * @return string
+	 */
+	function trim_quotes( $string ) {
+		return trim( $string, '"\'' );
+	}
+
+	/**
+	 * Processes this test, when one of its tokens is encountered.
+	 *
+	 * @param PHP_CodeSniffer_File $phpcsFile The file being scanned.
+	 * @param int                  $stackPtr  The position of the current token
+	 *                                        in the stack passed in $tokens.
+	 *
+	 * @return void
+	 */
+	public function process( PHP_CodeSniffer_File $phpcsFile, $stackPtr ) {
+
+		$tokens  = $phpcsFile->getTokens();
+		$token   = $tokens[ $stackPtr ];
+
+		$content = $this->trim_quotes( $token['content'] );
+
+		if ( 'wp_deregister_script' === $content ) {
+
```

Comment:
You're currently not checking whether this is actually a function call, so I think this will give false positives with code like the below. IMHO not a blocker for now as it'll be rare to encounter that kind of code.
Similar to the above mentioned utility function, there's also an _is this a function call ?_ utility function I intend to add to WPCS. Once that's in, we can start using that here and that should get rid of the false positives.

```php
My_Class::wp_deregister_scripts( 'jquery' );
$my_object->wp_deregister_scripts( 'jquery' );

class My_Class {
	function some_function( $var = 'jquery' ) {}
}
```